src/overlays/actors/ovl_En_Rat/z_en_rat.c

```diff
@@ -28,123 +44,910 @@ const ActorInit En_Rat_InitVars = {
+    if (bomb != NULL) {
+        bomb->timer = 0;
+    }
+
+    if (EN_RAT_GET_TYPE(&this->actor) == EN_RAT_TYPE_OVERWORLD) {
+        Item_DropCollectibleRandom(play, &this->actor, &this->actor.world.pos, 0x100);
+    }
+
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = EnRat_PostDetonation;
+}
+
+/**
+ * If the Real Bombchu is an Overworld-type Bombchu, this will set it up to revive.
+ * Otherwise, it will simply kill the Bombchu.
+ */
+void EnRat_PostDetonation(EnRat* this, PlayState* play) {
+    if (EN_RAT_GET_TYPE(&this->actor) == EN_RAT_TYPE_OVERWORLD) {
+        EnRat_SetupRevive(this);
+    } else {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnRat_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnRat* this = THIS;
+
+    this->shouldRotateOntoSurfaces = false;
+    if (this->stunTimer == 0) {
+        SkelAnime_Update(&this->skelAnime);
+    }
+
+    if (this->collider.base.atFlags & AT_HIT) {
+        this->collider.base.atFlags &= ~AT_HIT;
+        this->collider.base.acFlags &= ~AC_HIT;
+        this->collider.base.ocFlags1 &= ~OC1_HIT;
+        if (this->collider.base.atFlags & AT_BOUNCED) {
+            EnRat_SetupBounced(this);
+        } else {
+            EnRat_Explode(this, play);
+            return;
+        }
+    } else if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        if (this->actor.colChkInfo.damageEffect == EN_RAT_DMGEFF_HOOKSHOT) {
+            this->stunTimer = -2;
+        } else if (this->actor.colChkInfo.damageEffect == EN_RAT_DMGEFF_STUN) {
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_COMMON_FREEZE);
+            Actor_SetColorFilter(&this->actor, 0, 120, 0, 40);
+            if (this->actionFunc == EnRat_Bounced) {
+                this->actor.speedXZ = 0.0f;
+                if (this->actor.velocity.y > 0.0f) {
+                    this->actor.velocity.y = 0.0f;
+                }
+            } else {
+                this->stunTimer = 40;
+            }
+        } else {
+            EnRat_Explode(this, play);
+            return;
+        }
+    } else if (((this->collider.base.ocFlags1 & OC1_HIT) && (((this->collider.base.oc->category == ACTORCAT_ENEMY)) ||
+                                                             (this->collider.base.oc->category == ACTORCAT_BOSS) ||
+                                                             (this->collider.base.oc->category == ACTORCAT_PLAYER))) ||
+               ((this->actionFunc == EnRat_ChasePlayer) && (this->timer == 0))) {
+        this->collider.base.ocFlags1 &= ~OC1_HIT;
+        EnRat_Explode(this, play);
+        return;
+    }
+
+    this->actionFunc(this, play);
+
+    if ((this->actionFunc != EnRat_PostDetonation) && (this->actionFunc != EnRat_Revive)) {
+        if (this->stunTimer > 0) {
+            this->stunTimer--;
+        } else if (this->stunTimer < 0) {
+            if (this->stunTimer == -2) {
+                this->stunTimer = -1;
```

Comment:
Hmm, non timer behavior on timer. Union?

Reply:
Ehhh, not really convinced. When you hookshot a Bombchu, it sets `stunTimer` to -2, which then *immediately* goes to -1 (this is the code you highlighted). However, they don't *do* anything with that value; they never check it later in the actor, making the whole thing sort of pointless. I didn't feel like making a union for something that might as well not exist.

Reply:
If you are sure -1/-2 does nothing should you include a comment here saying the values aren't used by the code for anything? My opinion here is to leave some sort of comment to stop the reader from thinking those values actually have purpose and understanding them isn't necessary nearby to prevent confusion, although I have been asked to remove comments like that before.

Reply:
Reading it again and cross-referencing with OoT's flags documentation, it actually does *something*, but it's a bit silly. They start by setting it to -2, and I think after this point, the Bombchu gets hooked (aka, ACTOR_FLAG_2000 is set). Then, it gets set to -1, and any time *after* this, if ACTOR_FLAG_2000 is unset, the Bombchu blows up.

I guess you could call this something like `hookedState` or something, with values like `HOOK_STARTED` and `HOOKED`. But I'm still not 100% sure it deserves a union. What do you think?

Reply:
Yeah I would union at that point, with the two states, I don't have a better name for the two states although I like hookedState.